apps/frontend/src/pages/auth/libs/components/reset-password-form/reset-password-form.tsx

```diff
@@ -54,6 +73,18 @@ const ResetPasswordForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
 		setIsConfirmPasswordVisible((previousState) => !previousState);
 	}, []);
 
+	const dispatch = useAppDispatch();
+	const navigate = useNavigate();
+
+	useEffect(() => {
+		void dispatch(authActions.checkIsResetPasswordExpired({ token }));
+	}, [dispatch, token]);
+
+	if (redirectLink === AppRoute.SIGN_IN) {
+		navigate(redirectLink);
+		dispatch(appActions.changeLink(null));
+	}
```

Comment:
This should be in a global file, not in auth. We can have other redirect in other places. Also no need to check if it is SIGN_IN route